newrelic/data_source_newrelic_key_transaction.go

```diff
@@ -28,37 +46,43 @@ func dataSourceNewRelicKeyTransactionRead(ctx context.Context, d *schema.Resourc
 	client := meta.(*ProviderConfig).NewClient
 
 	log.Printf("[INFO] Reading New Relic key transactions")
-
-	name := d.Get("name").(string)
-
-	params := apm.ListKeyTransactionsParams{
-		Name: name,
+	// fetch name from TF config
+	name, nameOk := d.GetOk("name")
+	if !nameOk {
+		return diag.FromErr(errors.New("`name` is required"))
+	}
+	// fetch guid from TF config
+	guid := d.Get("guid")
+	query := fmt.Sprintf("type = 'KEY_TRANSACTION' AND name = '%s'", name)
+	if guid != "" {
+		// if guid is provided, add it to the query to filter the results
+		query = fmt.Sprintf("%s AND id = '%s'", query, guid)
 	}
 
-	transactions, err := client.APM.ListKeyTransactionsWithContext(ctx, &params)
+	transaction, err := client.Entities.GetEntitySearchByQueryWithContext(ctx, entities.EntitySearchOptions{}, query, []entities.EntitySearchSortCriteria{})
```

Comment:
```suggestion
	keyTransactionsFound, err := client.Entities.GetEntitySearchByQueryWithContext(ctx, entities.EntitySearchOptions{}, query, []entities.EntitySearchSortCriteria{})
```

The variable can be renamed as this is no longer a collection of `*apm.KeyTransaction` as in REST v2

Reply:
updated